### ISSUE TITLE
[SODA] Tooltip content not accessible by screen readers.

### ISSUE THREAD

**Issue by aaron-collier** [sul-dlss/hungry-hungry-hippo#898]
Tooltip content is not accessible to screen readers. The presence of the tooltip is not conveyed to  screen reader users and it cannot be accessed by keyboard users.

SODA Recommendation:

```
RULE : 
The name, role, value, states, and properties of user interface components MUST be programmatically determinable by assistive technologies.

HOW TO FIX: 
A tooltip is a popup that displays information related to an element. There are several common patterns for tooltips including tooltips that appear on mouse hover or keyboard focus of the tooltip trigger and tooltips that appear on activation of the tooltip trigger. Tooltips may contain only static text or they may contain functional elements.

Fix this issue by ensuring all of the following are true:
1. The tooltip trigger must be focusable and operable with a keyboard.
2. If the tooltip content contains functional elements - such as links, buttons or form fields - they must be focusable and operable with a keyboard.
3. The content of the tooltip must be readable with a screen reader.
4. The keyboard focus (Tab key) order and screen reader reading order must be logical. The simplest way to ensure this is to insert the tooltip content immediately after the tooltip trigger in the DOM.

REFERENCE: 
Deque University: https://dequeuniversity.com/class/custom-widgets/examples/tooltip
Deque University: https://dequeuniversity.com/class/custom-widgets/examples/tooltip-dialog
WAI-ARIA Authoring Practices: https://www.w3.org/WAI/ARIA/apg/patterns/tooltip/

BACKGROUND: 
Every user interface control must have a role along with any applicable states and properties so that screen reader users know how to interact with the control. Native HTML elements - such as <button>, <a>, <input>, <select> - already have a role, and their necessary states and properties - such as the checked/unchecked state of a checkbox - are automatically conveyed so nothing more needs to be done. If you create a custom version of a native HTML element or a custom control or widget that does not have a native HTML equivalent, you must add the relevant role(s) and any applicable states and properties using ARIA as well as expected keyboard interactions.
```

See Issue ID 2252 in https://docs.google.com/spreadsheets/d/1x0uBBDe3T8O0HNCs74OdVKQc8tQQNvOV_f-C6IS44IQ/edit?gid=1050712151#gid=1050712151

**Comment by thatbudakguy** [comment]
After investigating with a screen reader, I've found that it does not read the content of the tooltip, and the SR itself does not focus the tooltip when moving through the page (it is only focusable via tab).

Looking [at Bootstrap's docs](https://getbootstrap.com/docs/5.3/components/tooltips/#markup), they describe exactly this situation:

> Keep tooltips accessible to keyboard and assistive technology users by only adding them to HTML elements that are traditionally keyboard-focusable and interactive (such as links or form controls). **While other HTML elements can be made focusable by adding tabindex="0", this can create annoying and confusing tab stops on non-interactive elements for keyboard users, and most assistive technologies currently do not announce tooltips in this situation**. Additionally, do not rely solely on hover as the trigger for your tooltips as this will make them impossible to trigger for keyboard users.

This is our problem: our tooltips are not interactive elements. We are using the `<i>` tag to tie the tooltip to an icon, but this tag is not interactive (and arguably not appropriate for usage as an icon, but that's a whole other can of worms).

**Comment by thatbudakguy** [comment]
First thought: just make all tooltips `<button>` elements and keep the styling as it is. Since they require clicking in order to activate, they're effectively buttons already.

**Comment by thatbudakguy** [comment]
This doesn't work either; the user hears that there's a button but has no context for what it is or why they should activate it. 

This is unsurprising, given [MDN says](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role#description):

> The tooltip is not the appropriate role for the more information "i" icon, ⓘ. A tooltip is directly associated with the owning element. The ⓘ isn't 'described by' detailed information; the tool or control is.

So, we can't get away with componentizing tooltips – they always need to be tied to whatever they are describing, rather than tied to an icon that has no intrinsic meaning.

Bootstrap doesn't make this easy because its tooltips _don't exist_ until you hover/activate the item, at which point something is added to the DOM with `role=tooltip`. But a SR can't focus or activate the object if it isn't something like a button or link that could already be focused or activated, so it can't trigger the tooltip to be added to the DOM in order to be read.
